test/tap/tests/mysql_hostgroup_attributes_config_file-t.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file mysql_hostgroup_attributes_config_file-t.cpp
+ * @brief Reading and saving 'mysql_hostgroup_attributes' table from configuration file:
+ *   1. Correct values should be inserted into 'mysql_hostgroup_attributes' table.
+ *   2. Correct saving of mysql_hostgroup_attributes into config file.
+ *   3. Delete all the values from 'mysql_hostgroup_attributes' table
+ *   4. Correct load of 'mysql_hostgroup_attributes' table values from config file
+ */
+
+#include <cstring>
+#include <string>
+#include <fstream>
+#include <unistd.h>
+
+#include "mysql.h"
+#include "mysqld_error.h"
+
+#include "json.hpp"
+
+#include "tap.h"
+#include "utils.h"
+#include "command_line.h"
+
+using nlohmann::json;
+using std::string;
+using std::fstream;
+
+int save_and_read_mysql_hostgroup_attributes_from_config(MYSQL* admin) {
+	json j_servers_defaults { { "weight", 100 }, { "max_connections", 10000 }, { "use_ssl", 1}, };
+
+	// To run this test locally copy a ProxySQL config file into this path.
+	string save_config_query = {"SAVE CONFIG TO FILE /var/lib/jenkins//scripts/"
```

Comment:
I don't think TAP tests should modify any config file. Config file should be immutable once the testing starts, for the whole duration of the CI run.

Thus a different approach could be:
* add `mysql_hostgroup_attributes` in the default configuration file . ( @mirostauder ? )
* the TAP test could verify that the configuration already loaded (at the beginning of the CI run, not at the beginning of the TAP test itself) is correct
* wipe the current table
* issue a new `LOAD MYSQL SERVERS FROM CONFIG`
* verify that the configuration is again what expected
* issue a `SAVE CONFIG TO FILE` on the TAP test working directory
* verify that the `mysql_hostgroup_attributes` section in the file is correct . The tap test itself could have hardcoded what to expect in the `mysql_hostgroup_attributes` section of the file

Reply:
A possible enhancement (not strictly related to this PR) is that `LOAD MYSQL SERVERS FROM CONFIG` could accept a filename. And if this is the case, `Read_MySQL_Servers_from_configfile()` should accept a config file path.

Reply:
Thanks Rene. I will create a issue for the same.